File: src/main/java/com/eager/questioncloud/user/PasswordProcessor.java
package com.eager.questioncloud.user;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

public class PasswordProcessor {
    public static PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();

    public static String encode(String rawPassword) {
        return passwordEncoder.encode(rawPassword);
    }

    public static Boolean matches(String rawPassword, String encodedPassword) {
        return passwordEncoder.matches(rawPassword, encodedPassword);
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/user/implement/PasswordProcessor.kt
package com.eager.questioncloud.core.domain.user.implement

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

object PasswordProcessor {
    private val passwordEncoder: PasswordEncoder = BCryptPasswordEncoder()

    fun encode(rawPassword: String): String {
        return passwordEncoder.encode(rawPassword)
    }

    fun matches(rawPassword: String, encodedPassword: String): Boolean {
        return passwordEncoder.matches(rawPassword, encodedPassword)
    }
}
